hudi-hadoop-mr/src/main/java/org/apache/hudi/hadoop/realtime/RealtimeSplit.java

```diff
@@ -34,49 +36,73 @@
  */
 public interface RealtimeSplit extends InputSplitWithLocationInfo {
 
+
   /**
    * Return Log File Paths.
+   *
    * @return
    */
   List<String> getDeltaLogPaths();
 
   /**
    * Return Max Instant Time.
+   *
    * @return
    */
   String getMaxCommitTime();
 
   /**
    * Return Base Path of the dataset.
+   *
    * @return
    */
   String getBasePath();
 
+  /**
+   * Returns Virtual key info if meta fields are disabled.
+   * @return
+   */
+  Option<HoodieVirtualKeyInfo> getHoodieVirtualKeyInfoOpt();
+
   /**
    * Update Log File Paths.
+   *
    * @param deltaLogPaths
    */
   void setDeltaLogPaths(List<String> deltaLogPaths);
 
   /**
    * Update Maximum valid instant time.
+   *
    * @param maxCommitTime
    */
   void setMaxCommitTime(String maxCommitTime);
 
   /**
    * Set Base Path.
+   *
    * @param basePath
    */
   void setBasePath(String basePath);
 
+  void setHoodieVirtualKeyInfoOpt(Option<HoodieVirtualKeyInfo> hoodieVirtualKeyInfoOpt);
+
   default void writeToOutput(DataOutput out) throws IOException {
     InputSplitUtils.writeString(getBasePath(), out);
     InputSplitUtils.writeString(getMaxCommitTime(), out);
     out.writeInt(getDeltaLogPaths().size());
     for (String logFilePath : getDeltaLogPaths()) {
       InputSplitUtils.writeString(logFilePath, out);
     }
+    if (!getHoodieVirtualKeyInfoOpt().isPresent()) {
+      InputSplitUtils.writeString(HoodieRealtimeInputFormatUtils.NULL_STR, out);
```

Comment:
@vinothchandar : also, is there some other elegant way to do this. I can do a boolean. but not sure if that has any major benefit over this. In InputSpltiUtils, we have apis for string. And so chose this way.

Reply:
I ll let you make the call here. it just boils down to efficiency of storage.